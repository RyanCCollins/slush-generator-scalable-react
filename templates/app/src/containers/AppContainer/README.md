## AppContainer
The top level app component that sits on top of other components.

### Example Usage

```js
<AppContainer  />
```


### Other Information
Store global state, such as an auth token or a user here.
