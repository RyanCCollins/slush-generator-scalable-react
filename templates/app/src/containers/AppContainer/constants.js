export const APPCONTAINER_DEFAULT_ACTION = 'APPCONTAINER_DEFAULT_ACTION';
