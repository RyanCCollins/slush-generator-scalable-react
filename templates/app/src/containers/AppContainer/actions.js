import * as types from './constants';

// appContainerdefaultAction :: None -> {Action}
export const appContainerDefaultAction = () => ({  // eslint-disable-line
  type: types.APPCONTAINER_DEFAULT_ACTION,
});
