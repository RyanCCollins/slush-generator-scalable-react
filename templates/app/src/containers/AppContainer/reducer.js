import * as types from './constants';

export const initialState = {
  isMobile: false,
};

const appContainerReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.DEFAULT_ACTION:
        return state;
      default:
        return state;
    }
  };

export default appContainerReducer;
