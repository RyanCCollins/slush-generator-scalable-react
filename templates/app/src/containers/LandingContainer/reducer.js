import update from 'react-addons-update';
import * as types from './constants';

export const initialState = {
  isLoading: false,
  name: null,
  isShowingModal: false,
  contributors: [
    {
      name: 'Ryan Collins',
      github: 'ryanccollins',
      bio: 'Experienced Software Engineer specializing in implementing cutting-edge ' +
        'technologies in a multitude of domains, focusing on React and Front End.  ' +
        'Weekend Data Scientist and Functional Programmer.',
      avatar: 'https://avatars.githubusercontent.com/u/13810084?v=3',
    },
    {
      name: 'Andreas Daiminger',
      github: 'adai183',
      bio: 'I started to code about 2 years ago and have been very ' +
        'focused on developing cutting edge UI components.' +
        'I have a passion for functional programming, and I love creating ' +
        'environments that make it easier for developers to write consistent, testable code.',
      avatar: 'https://avatars.githubusercontent.com/u/13679375?v=3',
    },
  ],
};

const landingReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.OPEN_MODAL:
        return update(state, {
          isShowingModal: {
            $set: true,
          },
        });
      case types.CLOSE_MODAL:
        return update(state, {
          isShowingModal: {
            $set: false,
          },
        });
      case types.LOAD_DATA_INITIATION:
        return update(state, {
          isShowingModal: {
            $set: false,
          },
          isLoading: {
            $set: true,
          },
          name: {
            $set: action.name,
          },
        });
      case types.LOAD_DATA_SUCCESS:
        return update(state, {
          isLoading: {
            $set: false,
          },
        });
      default:
        return state;
    }
  };

export default landingReducer;
