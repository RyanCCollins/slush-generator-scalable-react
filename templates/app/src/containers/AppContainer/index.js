import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import App from 'grommet-udacity/components/App';
import { Navbar, AppFooter } from 'components';
import * as AppContainerActionCreators from './actions';

class AppContainer extends Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      location,
    } = this.props;
    return (
      <App centered={false} inline>
        <Navbar pathname={location.pathname} />
        {React.cloneElement(this.props.children, this.props)}
        <AppFooter />
      </App>
    );
  }
}

AppContainer.propTypes = {
  children: PropTypes.node.isRequired,
  location: PropTypes.object.isRequired, // eslint-disable-line
};

// mapStateToProps :: {State} -> {Props}
const mapStateToProps = state => ({
  isMobile: state.app.isMobile, // example / unused
});

// mapDispatchToProps :: Dispatch -> {Action}
const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    AppContainerActionCreators,
    dispatch,
  ),
});

const Container = AppContainer;

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Container);
