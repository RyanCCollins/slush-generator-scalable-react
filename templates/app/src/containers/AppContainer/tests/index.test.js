import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState as app } from '../reducer';
import AppContainer from '../index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<AppContainer />', () => {
  it('should render with default props', () => {
    const store = mockStore({ app });
    const wrapper = shallow(
      <AppContainer location={{ pathname: '/' }} store={store}>
        <div />
      </AppContainer>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot(); // eslint-disable-line
  });
});
