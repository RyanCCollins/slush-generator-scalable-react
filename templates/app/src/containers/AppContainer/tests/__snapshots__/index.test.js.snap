exports[`<AppContainer /> should render with default props 1`] = `
<AppContainer
  actions={
    Object {
      "appContainerDefaultAction": [Function],
    }
  }
  isMobile={false}
  location={
    Object {
      "pathname": "/",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }>
  <div />
</AppContainer>
`;
